python/lsst/obs/base/cameraMapper.py

```diff
@@ -1059,6 +1074,156 @@ def getRegistry(self):
+        """
+        recipeName = self.mappings[datasetType].recipe
+        if recipeName not in self._writeRecipes:
+            raise RuntimeError("Unrecognized write recipe for datasetType %s: %s" % (datasetType, recipeName))
+        recipe = self._writeRecipes[recipeName].deepCopy()
+        seed = hash(tuple(dataId.items())) % 2**31
+        for plane in ("image", "mask", "variance"):
+            if recipe.exists(plane + ".scaling.seed") and recipe.get(plane + ".scaling.seed") == 0:
+                recipe.set(plane + ".scaling.seed", seed)
+        return recipe
+
+    def _initWriteRecipes(self):
+        """Read the recipes for writing files
+
+        These recipes are currently used for configuring FITS compression,
+        but they could have wider uses for configuring different flavors
+        of the storage types. A recipe is referred to by a symbolic name,
+        which has associated settings. These settings are stored as a
+        `PropertySet` so they can easily be passed down to the
+        boost::persistence framework as the "additionalData" parameter.
+
+        The list of recipes is written in YAML. A default recipe and
+        some other convenient recipes are in obs_base/policy/writeRecipes.yaml
+        and these may be overridden or supplemented by the individual obs_*
+        packages' own policy/writeRecipes.yaml files.
+
+        Each recipe for FITS compression should define "image", "mask" and
+        "variance" entries, each of which may contain "compression" and
+        "scaling" entries. Defaults will be provided for any missing
+        elements under "compression" and "scaling".
+
+        The allowed entries under "compression" are:
+
+        * scheme (string): compression algorithm to use
+        * rows (int): number of rows per tile (0 = entire image)
+        * quantizeLevel (float): cfitsio quantization level
+
+        The allowed entries under "scaling" are:
+
+        * scheme (string): scaling algorithm to use
+        * bitpix (int): bits per pixel (0,8,16,32,64,-32,-64)
+        * fuzz (bool): fuzz the values when quantising floating-point values?
+        * seed (long): seed for random number generator when fuzzing
+        * maskPlanes (list of string): mask planes to ignore when doing statistics
+        * quantizeLevel: divisor of the standard deviation for STDEV_* scaling
+        * quantizePad: number of stdev to allow on the low side (for STDEV_POSITIVE/NEGATIVE)
+        * bscale: manually specified BSCALE (for MANUAL scaling)
+        * bzero: manually specified BSCALE (for MANUAL scaling)
+
+        A very simple example YAML recipe:
+
+            default:
+              image: &default
+                compression:
+                  scheme: GZIP_SHUFFLE
+              mask: *default
+              variance: *default
+
+        This method raises a `RuntimeError` if the YAML contents don't
+        match the expected schema.
+        """
```

Comment:
this info needs to be in LDM-463, in the CameraMapper section. It would be good to also add a high level description of that Recipes are and what uses they are intended for.